subsys/bluetooth/controller/include/ll.h

```diff
@@ -83,6 +83,8 @@ u8_t ll_terminate_ind_send(u16_t handle, u8_t reason);
 u8_t ll_rssi_get(u16_t handle, u8_t *rssi);
 u8_t ll_tx_pwr_lvl_get(u16_t handle, u8_t type, s8_t *tx_pwr_lvl);
 void ll_tx_pwr_get(s8_t *min, s8_t *max);
+void ll_tx_pwr_set(const u32_t tx_pwr_lvl);
```

Comment:
Enhance this with BLE state/role (Adv, Scan, Connection) type, handle per state/role. Could ll_tx_pwr_lvl_get (from HCI) be used as inspiration?

Reply:
@cvinayak thanks for the review and the suggestions! I started applying them, and reached the current solution aiming to solve most of your points:
- expand the LL setter to include LL node role (ADV, SCAN, CONN) and handles as 
```ll_tx_pwr_lvl_set(enum ll_state u16_t handle, u8_t type, s8_t *tx_pwr_lvl);```
- declare in ```ll.h``` the interfaces 
```
enum ll_state {
	ADV,
	SCAN,
	CONN
};

void ll_adv_tx_pwr_lvl_set(u16_t handle, const s8_t tx_pwr_lvl);
void ll_scan_tx_pwr_lvl_set(u16_t handle, const s8_t tx_pwr_lvl);
void ll_conn_tx_pwr_lvl_set(u16_t handle, const s8_t tx_pwr_lvl);
```
that need to be implemented by LL_SW legacy and split versions to provide the dynamic tx control 
- for LL_SW split I have the call path: LL -> ULL -> LLL -> radio -> hal_radio and the logic is now handled at ULL level where the above 3 methods are implemented and at LLL to actually modify the radio Tx in protected BT events
- for LL_SW legacy I am trying to implement now the logic at the controller level decoupling the paths as you mentioned between the roles; however, I saw that atm there are no multiple handles for ADV and SCAN roles; since this path is going to become deprecated I believe than I don't have to add handling of individual ADV/SCAN and just work with the already global advertiser and scanner instances from the radio struct, e.g. https://github.com/zephyrproject-rtos/zephyr/blob/fca32e41e608dd4751df2e7e664f1cce70b37998/subsys/bluetooth/controller/ll_sw/ctrl.c#L195 and 
https://github.com/zephyrproject-rtos/zephyr/blob/fca32e41e608dd4751df2e7e664f1cce70b37998/subsys/bluetooth/controller/ll_sw/ctrl.c#L196
would this be acceptable ?
- the rest of the work then is to simply create a VS HCI command and implement some examples and tests.

Any comments from your side ?

Reply:
I agree with most. Thumbs up towards the feature enhancements you are doing :-)

Instead of three new interface just change:
```u8_t ll_tx_pwr_lvl_get(u16_t handle, u8_t type, s8_t *tx_pwr_lvl);``` to ```u8_t ll_tx_pwr_lvl_get(enum ll_state state, u16_t handle, u8_t type, s8_t *tx_pwr_lvl);```
In the hci.c where it is used, pass CONN as the state, and in new vendor HCI interface pass state as supplied by application.

and add:
```u8_t ll_tx_pwr_lvl_set(enum ll_state state, u16_t handle, s8_t tx_pwr_lvl);``` for dynamic control.

For the legacy controller, ignore the handle supplied and directly reference the singular advertiser and scanner structs.